Lib/pdb.py

```diff
@@ -2435,30 +2435,38 @@ def main():
     parser.add_argument('-c', '--command', action='append', default=[], metavar='command', dest='commands',
                         help='pdb commands to execute as if given in a .pdbrc file')
     parser.add_argument('-m', metavar='module', dest='module')
-    parser.add_argument('args', nargs='*',
-                        help="when -m is not specified, the first arg is the script to debug")
 
     if len(sys.argv) == 1:
         # If no arguments were given (python -m pdb), print the whole help message.
         # Without this check, argparse would only complain about missing required arguments.
         parser.print_help()
         sys.exit(2)
 
-    opts = parser.parse_args()
+    opts, args = parser.parse_known_args()
+
+    invalid_args = []
+    for arg in args:
+        if not arg.startswith('-'):
+            break
```

Comment:
What's the logic here? Why are we breaking?

Reply:
A pdb CLI usage could be `python -m pdb script.py --bar` or `python -m pdb --bar script.py`.

The unknown args would be `['script.py', '--bar']` and `[--bar', 'script.py']` respectively. Notice that in the first case, the user is trying to pass `--bar` to `script.py`, but in the latter, they try to pass `--bar` to `pdb`. This logic is to differentiate this. As long as we got something that does not start with `-`, we consider that as the script.

Reply:
I'm missing something here, for the module case.

Reply:
> I'm missing something here, for the module case.

And a test for this case..

Reply:
Turns out dealing with modules is not super trivial. I added some comments in the parsing code and hopefully that's helpful. We can "change the rule" and require users to use `--` to resolve all ambiguities, but I think it would be a better experience if we can be a bit smarter.

An existing test already covers the case where `--spam -m module` should fail, I added a new test for `-m calendar --type text` which works after the change.